File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/client/ClientResponse.java
package org.hyperledger.iroha.android.client;

import java.util.Arrays;
import java.util.Objects;

/** Simple container for responses returned by {@link IrohaClient} implementations. */
public final class ClientResponse {
  private final int statusCode;
  private final byte[] body;
  private final String message;
  private final String transactionHashHex;
  private final String rejectCode;

  public ClientResponse(final int statusCode, final byte[] body) {
    this(statusCode, body, "", null, null);
  }

  public ClientResponse(final int statusCode, final byte[] body, final String message) {
    this(statusCode, body, message, null, null);
  }

  public ClientResponse(
      final int statusCode,
      final byte[] body,
      final String message,
      final String transactionHashHex) {
    this(statusCode, body, message, transactionHashHex, null);
  }

  public ClientResponse(
      final int statusCode,
      final byte[] body,
      final String message,
      final String transactionHashHex,
      final String rejectCode) {
    this.statusCode = statusCode;
    this.body = Arrays.copyOf(Objects.requireNonNull(body, "body"), body.length);
    this.message = Objects.requireNonNull(message, "message");
    this.transactionHashHex = transactionHashHex == null ? null : transactionHashHex;
    this.rejectCode = rejectCode == null || rejectCode.isBlank() ? null : rejectCode;
  }

  public int statusCode() {
    return statusCode;
  }

  public byte[] body() {
    return Arrays.copyOf(body, body.length);
  }

  public String message() {
    return message;
  }

  public java.util.Optional<String> hashHex() {
    return java.util.Optional.ofNullable(transactionHashHex);
  }

  public java.util.Optional<String> rejectCode() {
    return java.util.Optional.ofNullable(rejectCode);
  }
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/client/ClientResponse.kt
package org.hyperledger.iroha.sdk.client

/** Simple container for responses returned by `IrohaClient` implementations. */
class ClientResponse(
    @JvmField val statusCode: Int,
    body: ByteArray,
    @JvmField val message: String = "",
    transactionHashHex: String? = null,
    rejectCode: String? = null,
) {
    private val _body: ByteArray = body.copyOf()

    @JvmField val transactionHashHex: String? = transactionHashHex
    @JvmField val rejectCode: String? = if (rejectCode.isNullOrBlank()) null else rejectCode

    val body: ByteArray get() = _body.copyOf()

    fun hashHex(): String? = transactionHashHex

    fun rejectCode(): String? = rejectCode
}
